src/components/settings/SettingsSSOTab/index.jsx

```diff
@@ -21,22 +24,143 @@ const SettingsSSOTab = ({ enterpriseId, setHasSSOConfig }) => {
   const [existingProviderData, pdError, pdIsLoading] = useExistingProviderData(enterpriseId, refreshBool);
   const [showNewSSOForm, setShowNewSSOForm] = useState(false);
   const [showNoSSOCard, setShowNoSSOCard] = useState(false);
+  const { AUTH0_SELF_SERVICE_INTEGRATION } = features;
+  const [isOpen, open, close] = useToggle(false);
+
+  const newConfigurationButtonOnClick = async () => {
+    for (let i = 0; i < existingConfigs.length; i++) {
+      Promise.all(LmsApiService.updateEnterpriseSsoOrchestrationRecord(
+        { active: false, is_removed: true },
+        existingConfigs[i].uuid,
+      )).then(() => {
+        setRefreshBool(!refreshBool);
+        close();
+      });
+    }
+  };
 
   useEffect(() => {
-    let validConfigExists = false;
-    existingConfigs.forEach(config => {
-      if (config.was_valid_at) {
-        validConfigExists = true;
-      }
-    });
+    if (AUTH0_SELF_SERVICE_INTEGRATION) {
+      let validConfigExists = false;
+      validConfigExists = existingConfigs.some(config => config.validated_at);
+      setHasSSOConfig(validConfigExists);
+    } else {
+      let validConfigExists = false;
+      existingConfigs.forEach(config => {
+        if (config.was_valid_at) {
+          validConfigExists = true;
+        }
```

Comment:
Can condense this logic with array.some() like the block above

Reply:
yeah I considered it, just didn't want to touch existing implementations much but I guess it's best practice